src/Symfony/Component/Console/Attribute/Argument.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Attribute;
+
+use Symfony\Component\Console\Completion\CompletionInput;
+use Symfony\Component\Console\Completion\Suggestion;
+use Symfony\Component\Console\Exception\LogicException;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+
+#[\Attribute(\Attribute::TARGET_PARAMETER)]
+class Argument
+{
+    private ?int $mode = null;
+
+    /**
+     * Represents a console command <argument> definition.
+     *
+     * If unset, the `name`, `mode` and `default` values might be inferred from the parameter definition.
+     *
+     * @param string|bool|int|float|array|null                               $default         The default value (for InputArgument::OPTIONAL mode only)
+     * @param array|callable-string(CompletionInput):list<string|Suggestion> $suggestedValues The values used for input completion
+     */
+    public function __construct(
+        public string $name = '',
+        public string $description = '',
+        public string|bool|int|float|array|null $default = null,
+        public array|string $suggestedValues = [],
```

Comment:
In PHP 8.5, closures will be allowed in attribute constant ([approved RFC](https://wiki.php.net/rfc/closures_in_const_expr)). It may be a bit early to add support here, but that's okay. Also, I don't see the point of allowing any string: if it's not a function name, this will create a type error line 79.
```suggestion
        public array|\Closure $suggestedValues = [],
```

Reply:
Yeah, we can’t use `\Closure` for now, nor `callable`, so I have no choice but to accept a string (plus a doc comment `callable-string(CompletionInput):list<string|Suggestion>`).

Reply:
Added a TypeError exception

Reply:
this will support object method calls until `\Closure` is supported: https://github.com/symfony/symfony/pull/59340/files#r1900541811